components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -424,6 +509,14 @@ private OAuthConsumerAppDTO createOAuthApp(ApplicationRegistrationRequest regist
         return createdApp;
     }
 
+    private boolean validateTokenEndpointAuthMethod(String tokenEndpointAuthMethod) {
+        if (Arrays.asList(TOKEN_ENDPOINT_AUTH_METHODS).contains(tokenEndpointAuthMethod)) {
```

Comment:
let's simplify this as `return Arrays.asList(TOKEN_ENDPOINT_AUTH_METHODS).contains(tokenEndpointAuthMethod)`

Reply:
Added null check of tokenEndpointAuthMethod to this function